### DISCUSSION TITLE
Negative Jocobian while reading mesh files

### DISCUSSION THREAD

**Discussion by leoforeal** [idaholab/moose#26136] [Q&A Meshing] [answered]
Hallo, 

I have a general question, what could be the possible defects of a mesh when encountering the error:
**ERROR: negative Jacobian -2.0939e+08 at point (x,y,z)=(4.44762e+06, 5.35074e+06, -483.696) in element 0**
while reading such mesh files for moose to run?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Is this an external mesh?

Negative Jacobian means the element is distorted in such a way that the Jacobian of the transformation between the real element shape and the reference shape/frame is negative. We error in this case as solutions would be unphysical

Guillaume

**Reply by GiudGiud** [reply]
Typical reasons are flipped elements. 

Can you use the MeshDiagnosticsGenerator after loading the mesh and try the various checks?

**Reply by leoforeal** [reply]
yes, thanks for the feedback!
it's a file mesh, and I multiply one of the coordinate by negative values, hence leading to Negative Jacobian.
I re-arranged the order element ids and now it works.
